packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1732,6 +1732,35 @@ abstract class MultiChildRenderObjectWidget extends RenderObjectWidget {
   /// If this list is going to be mutated, it is usually wise to put [Key]s on
   /// the widgets, so that the framework can match old configurations to new
   /// configurations and maintain the underlying render objects.
+  ///
+  /// Here's a common example where one mutates the children but forgets to set
+  /// a different key. In such case, the mutated children may not be reflected.
+  /// ```
+  /// class SomeWidgetState extends ... {
+  ///     List<Widget> _children;
+  ///
+  ///     void initState(...) {
+  ///         _children = [...];
+  ///     }
+  ///
+  ///     void someHandler() {
+  ///         setState(() {
+  ///             _children.add(...);
+  ///         });
+  ///     }
+  ///
+  ///     Widget build(...) {
+  ///         // A changing key is needed if someHandler modifies _children.
+  ///         return Row(children: _children/*, key: missing_key*/);
```

Comment:
woah no, that won't work. you'll be blowing away the entire Row each time if you do this, that's very expensive.

You want to set the keys on each of the _children, and you want to very explicitly _not_ have a `List` that you create outside of the `build`.